### PR TITLE
Add endpoint for getting correct Package name for given import name

### PR DESCRIPTION
**Pull request by tlegen-k**
## Related Issues and Dependencies
https://github.com/thoth-station/storages/pull/2410
https://github.com/thoth-station/datasets/pull/35
…

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

UserAPI endpoint to get python package name for given import name.

### CODE REVIEW

**Comment by tlegen-k** [comment]
Hi @pacospace 
Can you take a look on these changes?

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -1080,6 +1080,35 @@ paths:
               schema:
                 $ref: "#/components/schemas/PythonPackageMetadataResponseError"
 
+  /python/get-imported-packages:
```
Comment (pacospace) [inline openapi/openapi.yaml:1083]: the endpoint actually does the opposite, `/get-package-from-imports` cc @fridex

File: openapi/openapi.yaml [hunk 2]
```diff
@@ -1080,6 +1080,35 @@ paths:
...
+    get:
+      tags: [PythonPackages]
+      x-openapi-router-controller: thoth.user_api.api_v1
+      operationId: get_imported_packages
```
Comment (pacospace) [inline openapi/openapi.yaml:1087]: get_package_from_imported_packages

File: openapi/openapi.yaml [hunk 3]
```diff
@@ -1080,6 +1080,35 @@ paths:
...
+      summary: List imported packages' correct names.
+      parameters:
+        - name: name
```
Comment (pacospace) [inline openapi/openapi.yaml:1090]: import?

File: thoth/user_api/api_v1.py [hunk 4]
```diff
@@ -1139,3 +1139,23 @@ def _do_get_image_metadata(
         error_str = str(exc)
 
     return {"error": error_str, "parameters": locals()}, status_code
+
+
+def get_imported_packages(name: str):
```
Comment (pacospace) [inline thoth/user_api/api_v1.py:1144]: `import_package_name`?

File: openapi/openapi.yaml (hunk 1 above)
Comment (tlegen-k) [reply openapi/openapi.yaml]: fixed in 0ada6fd042629810b21dcec94b4be45bd7b93e6b

File: openapi/openapi.yaml (hunk 2 above)
Comment (tlegen-k) [reply openapi/openapi.yaml]: > get_package_from_imported_packages

fixed in 0ada6fd042629810b21dcec94b4be45bd7b93e6b

File: openapi/openapi.yaml (hunk 3 above)
Comment (tlegen-k) [reply openapi/openapi.yaml]: > import?

fixed in 0ada6fd042629810b21dcec94b4be45bd7b93e6b

File: thoth/user_api/api_v1.py (hunk 4 above)
Comment (tlegen-k) [reply thoth/user_api/api_v1.py]: > `import_package_name`?

fixed in 0ada6fd042629810b21dcec94b4be45bd7b93e6b

File: openapi/openapi.yaml [hunk 5]
```diff
@@ -1080,6 +1080,35 @@ paths:
               schema:
                 $ref: "#/components/schemas/PythonPackageMetadataResponseError"
 
+  /python/get-package-from-imports:
```
Comment (fridex) [inline openapi/openapi.yaml:1083]: get is already in the `GET` method. Might be a good idea to rename this to something like `/python/imports`.

File: openapi/openapi.yaml [hunk 6]
```diff
@@ -1080,6 +1080,35 @@ paths:
...
+    get:
+      tags: [PythonPackages]
+      x-openapi-router-controller: thoth.user_api.api_v1
+      operationId: get_package_from_imported_packages
+      summary: List imported packages' correct names.
```
Comment (fridex) [inline openapi/openapi.yaml:1088]:
```suggestion
      summary: List imported packages' names.
```

File: openapi/openapi.yaml [hunk 7]
```diff
@@ -1080,6 +1080,35 @@ paths:
...
+      parameters:
+        - name: import_name
+          in: query
+          required: true
+          description: >
+            Names of the Python Packages for the given import package name.
+          schema:
+            type: string
+            default: "tensorflow"
```
Comment (fridex) [inline openapi/openapi.yaml:1097]: Might be a good idea to show the sklearn/scikit-learn example. Ex. `sklearn.cluster.MiniBatchKMeans` on the input and `scikit-learn` in the response example (or any other package where package name does not correspon with the import).

**Review by fridex — COMMENTED** [review]
lgtm, thanks @tlegen-k for the work 👍🏻

**Comment by tlegen-k** [comment]
> lgtm, thanks @tlegen-k for the work 👍🏻

my pleasure, thank you for help!

**Comment by fridex** [comment]
Verified this works flawlessly. Opened https://github.com/thoth-station/storages/pull/2426 so that 404 is reported if no package is matching the given import. We can have this in a new thoth-storages module is released.

@tlegen-k nice job! 💯 👍🏻

**Comment by tlegen-k** [comment]
> Verified this works flawlessly. Opened [thoth-station/storages#2426](https://github.com/thoth-station/storages/pull/2426) so that 404 is reported if no package is matching the given import. We can have this in a new thoth-storages module is released.
> 
> @tlegen-k nice job! 💯 👍🏻

Awesome, thank you!

**Review by harshad16 — APPROVED** [review]
lgtm
thanks :100:

**Comment by pacospace** [comment]
this Pr requires thoth-station/storages v0.56.0, which was released, but it is not on PyPI @harshad16

**Comment by harshad16** [comment]
> this Pr requires thoth-station/storages v0.56.0, which was released, but it is not on PyPI 

Thanks @pacospace, i will take a look.

**Comment by pacospace** [comment]
> > this Pr requires thoth-station/storages v0.56.0, which was released, but it is not on PyPI
> 
> Thanks @pacospace, i will take a look.

no Thank you @harshad16!!

**Pull request merged**